### PR TITLE
 Add resource access controls to Input Validation Management API

### PR DESCRIPTION
**Pull request by kayathiri4**
### Proposed changes in this pull request
Add resource access controls to /api/server/v1/validation-rules REST API

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 1]
```diff
@@ -1355,6 +1355,15 @@
             <Scopes>internal_application_mgt_update</Scopes>
         </Resource>
 
+        <Resource context="(.*)/api/server/v1/validation-rules(.*)" secured="true" http-method="GET">
+            <Permissions>/permission/admin/manage/identity/claimmgt/metadata/view</Permissions>
+            <Scopes>internal_claim_meta_view</Scopes>
+        </Resource>
+        <Resource context="(.*)/api/server/v1/validation-rules(.*)" secured="true" http-method="PUT">
+            <Permissions>/permission/admin/manage/identity/applicationmgt/update</Permissions>
+            <Scopes>internal_application_mgt_update</Scopes>
```
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:1364]: shouldn't these be claim mgt metadata related permissions ?

Comment (kayathiri4) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml]: Addressed with d2b87068136eaacb185ffdb3117c07cbed2924fd

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 2]
```diff
@@ -2015,6 +2015,15 @@
             <Scopes>internal_application_mgt_update</Scopes>
         </Resource>
 
+        <Resource context="(.*)/api/server/v1/validation-rules(.*)" secured="true" http-method="GET">
+            <Permissions>/permission/admin/manage/identity/claimmgt/metadata/view</Permissions>
+            <Scopes>internal_claim_meta_view</Scopes>
+        </Resource>
+        <Resource context="(.*)/api/server/v1/validation-rules(.*)" secured="true" http-method="PUT">
+            <Permissions>/permission/admin/manage/identity/applicationmgt/update</Permissions>
+            <Scopes>internal_application_mgt_update</Scopes>
```
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:2024]: this also need to be changed

Comment (kayathiri4) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: Addressed with ddda99a95d67a4f21dc3d85cc7eda04c04b13995

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**